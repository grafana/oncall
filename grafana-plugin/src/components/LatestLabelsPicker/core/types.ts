export interface ItemRepresentation {
  id: string;
  repr: string;
}

export interface ItemGroup {
  key: ItemRepresentation;
  values: ItemRepresentation[];
}

export interface ItemSelected {
  key: ItemRepresentation;
  value: ItemRepresentation;
}