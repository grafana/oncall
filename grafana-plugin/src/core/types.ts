export interface ItemGroup {
  key: any;
  values: any[];
}

export interface ItemSelected {
  key: any;
  value: any;
}

export enum LabelInputType {
  key = 'key',
  value = 'value',
}

export interface ServiceLabelValidator {
  isValid: boolean;
  errorMessage?: string;
}
