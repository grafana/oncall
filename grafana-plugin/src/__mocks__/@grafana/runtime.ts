export const getBackendSrv = () => ({
  get: jest.fn(),
  post: jest.fn(),
});
