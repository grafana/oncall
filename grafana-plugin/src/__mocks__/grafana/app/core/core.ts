export const contextSrv = {
  hasRole: jest.fn(),
};
