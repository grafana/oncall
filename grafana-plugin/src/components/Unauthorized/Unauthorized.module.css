.not-found {
  margin: 50px auto;
  text-align: center;
  width: 400px;
}

.error-code {
  color: var(--warning-text-color);
}
