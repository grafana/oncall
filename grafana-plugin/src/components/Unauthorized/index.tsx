import React, { FC } from 'react';

// import cn from 'classnames/bind';

// import styles from './Unauthorized.module.css';

// const cx = cn.bind(styles);

const Unauthorized: FC = (_props) => <p>You don't have access!</p>;

export default Unauthorized;
