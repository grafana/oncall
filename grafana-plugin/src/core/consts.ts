// Not exported outside the library (unlike types.ts)

export const KEY_ERROR_MESSAGE =
  'Key field is not valid (must start and end with an alphanumeric and can contain underscores)';
export const VALUE_ERROR_MESSAGE =
  'Value field is not valid (must start and end with an alphanumeric, can contain dashes, dots and underscores)';
