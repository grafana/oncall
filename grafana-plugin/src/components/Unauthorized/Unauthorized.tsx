import React, { FC } from 'react';

import { OrgRole } from '@grafana/data';
import { VerticalGroup } from '@grafana/ui';
import cn from 'classnames/bind';
import { contextSrv } from 'grafana/app/core/core';

import { Text } from 'components/Text/Text';
import { UserAction } from 'utils/authorization/authorization';

import styles from './Unauthorized.module.css';

const cx = cn.bind(styles);

type Props = {
  requiredUserAction: UserAction;
};

export const Unauthorized: FC<Props> = ({ requiredUserAction: { permission, fallbackMinimumRoleRequired } }) => (
  <div className={cx('not-found')}>
    <VerticalGroup spacing="lg" align="center">
      <Text.Title level={1} className={cx('error-code')}>
        403
      </Text.Title>
      <Text.Title level={4}>
        You do not have access to view this page.{' '}
        {contextSrv.licensedAccessControlEnabled()
          ? `You are missing the ${permission} permission.`
          : `You must be at least a${
              fallbackMinimumRoleRequired === OrgRole.Viewer ? '' : 'n'
            } ${fallbackMinimumRoleRequired}.`}
        <br />
        <br />
        Please contact your organization administrator to request access.
      </Text.Title>
    </VerticalGroup>
  </div>
);
