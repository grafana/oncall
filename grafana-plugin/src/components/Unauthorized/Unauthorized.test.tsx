import React from 'react';

import { OrgRole } from '@grafana/data';
import { contextSrv } from 'grafana/app/core/core';
import renderer from 'react-test-renderer';

import { Unauthorized } from 'components/Unauthorized/Unauthorized';

jest.mock('grafana/app/core/core', () => ({
  contextSrv: {
    accessControlEnabled: (): boolean => null,
  },
}));

describe('Unauthorized', () => {
  test.each([true, false])('renders properly - access control enabled: %s', (accessControlEnabled) => {
    contextSrv.licensedAccessControlEnabled = () => accessControlEnabled;
    const tree = renderer
      .create(
        <Unauthorized
          requiredUserAction={{
            permission: 'grafana-oncall-app.testing:hi',
            fallbackMinimumRoleRequired: OrgRole.Admin,
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test.each([OrgRole.Admin, OrgRole.Editor, OrgRole.Viewer])(
    'renders properly the grammar for different roles - %s',
    (role) => {
      contextSrv.licensedAccessControlEnabled = () => false;
      const tree = renderer
        .create(
          <Unauthorized
            requiredUserAction={{
              permission: 'grafana-oncall-app.testing:hi',
              fallbackMinimumRoleRequired: role,
            }}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    }
  );
});
