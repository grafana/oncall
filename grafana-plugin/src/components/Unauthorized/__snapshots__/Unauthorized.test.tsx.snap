// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized renders properly - access control enabled: false 1`] = `
<div
  className="not-found"
>
  <div
    className="css-8tu8mo-vertical-group"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h1
        className="title error-code"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          403
        </span>
      </h1>
    </div>
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h4
        className="title"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          You do not have access to view this page.
           
          You must be at least an Admin.
          <br />
          <br />
          Please contact your organization administrator to request access.
        </span>
      </h4>
    </div>
  </div>
</div>
`;

exports[`Unauthorized renders properly - access control enabled: true 1`] = `
<div
  className="not-found"
>
  <div
    className="css-8tu8mo-vertical-group"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h1
        className="title error-code"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          403
        </span>
      </h1>
    </div>
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h4
        className="title"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          You do not have access to view this page.
           
          You are missing the grafana-oncall-app.testing:hi permission.
          <br />
          <br />
          Please contact your organization administrator to request access.
        </span>
      </h4>
    </div>
  </div>
</div>
`;

exports[`Unauthorized renders properly the grammar for different roles - Admin 1`] = `
<div
  className="not-found"
>
  <div
    className="css-8tu8mo-vertical-group"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h1
        className="title error-code"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          403
        </span>
      </h1>
    </div>
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h4
        className="title"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          You do not have access to view this page.
           
          You must be at least an Admin.
          <br />
          <br />
          Please contact your organization administrator to request access.
        </span>
      </h4>
    </div>
  </div>
</div>
`;

exports[`Unauthorized renders properly the grammar for different roles - Editor 1`] = `
<div
  className="not-found"
>
  <div
    className="css-8tu8mo-vertical-group"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h1
        className="title error-code"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          403
        </span>
      </h1>
    </div>
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h4
        className="title"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          You do not have access to view this page.
           
          You must be at least an Editor.
          <br />
          <br />
          Please contact your organization administrator to request access.
        </span>
      </h4>
    </div>
  </div>
</div>
`;

exports[`Unauthorized renders properly the grammar for different roles - Viewer 1`] = `
<div
  className="not-found"
>
  <div
    className="css-8tu8mo-vertical-group"
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h1
        className="title error-code"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          403
        </span>
      </h1>
    </div>
    <div
      className="css-qxdyop-layoutChildrenWrapper"
    >
      <h4
        className="title"
      >
        <span
          className="root text text--undefined text--medium"
          style={
            {
              "maxWidth": undefined,
            }
          }
        >
          You do not have access to view this page.
           
          You must be at least a Viewer.
          <br />
          <br />
          Please contact your organization administrator to request access.
        </span>
      </h4>
    </div>
  </div>
</div>
`;
